packages/ra-ui-materialui/src/input/DateTimeInput.tsx

```diff
@@ -14,8 +14,8 @@ const leftPad2 = leftPad(2);
  * @param {Date} value value to convert
  * @returns {String} A standardized datetime (yyyy-MM-ddThh:mm), to be passed to an <input type="datetime-local" />
  */
-const convertDateToString = value => {
-    if (!(value instanceof Date) || isNaN(value)) return '';
+const convertDateToString = (value: Date) => {
+    if (!(value instanceof Date) || isNaN(value.getDate())) return '';
```

Comment:
If value is type Date, can `isNaN(value.getDate())` be true?

Reply:
It can be an invalid date, so yes